### ISSUE TITLE
Compliance/Payment Summaries - External User can view Payment Summaries

### ISSUE THREAD

**Issue by zoeyli-46** [bcgov/cas-compliance#20]
#### Description:

As an operator, I can view a table showing my payment history to date for all my operations on the payment summary tab for each invoice (obligation, penalty).

- [Design in Figma](https://www.figma.com/proto/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?page-id=14307%3A1663&node-id=24063-2&node-type=frame&viewport=1759%2C-4811%2C0.18&t=ZKSU89c6Y9Y5J6gI-1&scaling=scale-down&content-scaling=fixed&starting-point-node-id=20107%3A1497&show-proto-sidebar=1)

#### Columns:

- [ ] Compliance Period
- [ ] Operation name
- [ ] Payment towards
- [ ] Invoice number
- [ ] Payment amount applied
- [ ] Outstanding balance 

#### Development Checklist:

- [x] API route/service exists for dashboard payments
- [x] Service pulls all payment data for an operator from elicensing
- [x] Service combines elicensing data & obps data to craft response to frontend
- [x] Service is tested
- [x] The payment-summaries page displays a table with a row for each payment record returned

Note: The payment table is not searchable/sortable on purpose. that will be handled in #193 when we can leverage django's queryset object functionality.

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

·**Definition of Done** (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

#### Dependencies

- Blocked by
- Blocking

**Comment by nanyangpro** [comment]
Link to design is updated in the ticket description

**Comment by fviduya** [comment]
I have tested this in the DEV environment, [here is the test document for reference](https://buttoninc-my.sharepoint.com/:w:/g/personal/francesca_button_is/ER2299sjRx9ChRx2P4KKYFIB7qrBIaQhwjwhwHvqXTx0mg?e=hblfot).

I can verify that:
- [x] As an operator, I can view a table showing my payment history to date for all my operations on the payment summary tab for each invoice (obligation, penalty).

There were some things I noticed:
- The `Outstanding balance` column does not show the correct balance. In my [second test](https://buttoninc-my.sharepoint.com/:w:/g/personal/francesca_button_is/ER2299sjRx9ChRx2P4KKYFIB7qrBIaQhwjwhwHvqXTx0mg?e=yOVNZT&nav=eyJoIjoiMTA3NjYxNDAxMSJ9), I made two payments for a single obligation record but the first payment had an outstanding balance of 0. @vesselak, should this be handled in another ticket or am I okay to create another ticket to tackle that? 
- The Payment Summaries page seems to be loading slower than the other parts of the app. cc: @dleard 
- The Monetary Payments do not show in the obligation record itself.

Moving to Pending PO for review of test and notes. Thank you!

**Comment by vesselak** [comment]
@fviduya I have reviewed the testing documentation and here are a couple of comments:

- "outstanding balance" comes from the invoice, which is a parent of all payments. the table shows all payments to all obligations, so you wont' necessarily see the payments for 1 invoice together unless you filter by that invoice number.  
-  I agree with the comment of slow loading. Should we create a ticket to address this? Load testing ticket possibly?
- The Monetary Payment in the obligation record were addressed in #98 so please review both tickets and confirm if bug ticket is required before I accept this one. Thanks

**Comment by fviduya** [comment]
Hi @vesselak

![Image](https://github.com/user-attachments/assets/cede6d3b-d70b-45d3-ba93-539858249961)

Just to clarify for the image above, I made two payments for the same obligation. The first record was a partial payment of 400k, then the second record was to cover for the remaining balance. Should the first record not have an outstanding balance equivalent to what I paid in the second record? 

Also, I created a bug ticket for the monetary payments grid not having the correct data (#198). And another ticket to explore the slowness in the Payment Summaries page (#199). I think this ticket is good to go since we have the other issues covered in other tickets.

**Comment by vesselak** [comment]
Reviewed the testing documentation and accepting ticket with the deficiencies noted in #198 and #199
